backend/src/main/java/com/redhat/cloud/notifications/db/EventResources.java

```diff
@@ -23,123 +34,144 @@
+                                                    includePayload ? event.getPayload() : null,
+                                                    actions
+                                            );
+                                        }).collect(Collectors.toList());
 
-            return query.getSingleResult();
+                                        return buildPage(entries, count, limit, offset);
+                                    });
+                        }
+                    });
         });
     }
 
-    private Optional<String> getOrderByCondition(String sortBy) {
+    private Optional<String> getOrderByCondition(String sortBy, boolean nativeQuery) {
         if (sortBy == null) {
             return Optional.of(" ORDER BY e.created DESC");
         } else {
             Matcher sortByMatcher = SORT_BY_PATTERN.matcher(sortBy);
             if (sortByMatcher.matches()) {
-                String sortField = getSortField(sortByMatcher.group(1));
+                String sortField = getSortField(sortByMatcher.group(1), nativeQuery);
                 String sortDirection = sortByMatcher.group(2);
-                return Optional.of(" ORDER BY " + sortField + " " + sortDirection + ", e.created DESC");
+                String result = " ORDER BY " + sortField + " " + sortDirection;
+                if (!sortField.equals("e.created")) {
+                    result += ", e.created DESC";
+                }
+                return Optional.of(result);
             } else {
                 return Optional.empty();
             }
         }
     }
 
-    private Uni<List<UUID>> getEventIds(String accountId, Set<UUID> bundleIds, Set<UUID> appIds, String eventTypeDisplayName,
-                                        LocalDate startDate, LocalDate endDate, Set<EndpointType> endpointTypes, Set<Boolean> invocationResults,
-                                        Integer limit, Integer offset, Optional<String> orderByCondition) {
-        return sessionFactory.withSession(session -> {
-            String hql = "SELECT e.id FROM Event e JOIN e.eventType et JOIN et.application a JOIN a.bundle b " +
-                    "WHERE e.accountId = :accountId";
+    private Mutiny.Query<Object[]> buildEventIdsAndCountQuery(Mutiny.Session session, String accountId, Set<UUID> bundleIds,
+                                                              Set<UUID> appIds, String eventTypeDisplayName, LocalDate startDate, LocalDate endDate, Set<EndpointType> endpointTypes,
+                                                              Set<Boolean> invocationResults, Integer limit, Integer offset, String sortBy) {
 
-            hql = addHqlConditions(hql, bundleIds, appIds, eventTypeDisplayName, startDate, endDate, endpointTypes, invocationResults);
+        String sql = "SELECT CAST(e.id AS VARCHAR), COUNT(*) OVER() FROM event e, event_type et, applications a, bundles b " +
```

Comment:
`CAST(e.id AS VARCHAR)` is a workaround for an Hibernate Reactive issue/bug...

Reply:
I don't have a link and that workaround came from my own attempts to make the query work. Hopefully Hibernate ORM will work better and this will go away soon.